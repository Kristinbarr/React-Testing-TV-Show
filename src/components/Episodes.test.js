import React from 'react'
// import renderer from "react-test-renderer"; // 1: install this npm module as a dev dependency
import { render, fireEvent } from "@testing-library/react";

import Episodes from "./Episodes"

describe("<Episodes /> snapshot", () => {
  it("matches snapshot", () => {
    const tree = render(<Episodes />) // generates a DOM tree
    // snapshots are a JSON representation of the DOM tree
    expect(tree.asFragment()).toMatchSnapshot();
  });

  // Implementation details ->
  //   - what props.speak is doing...
  //   - if props.speak is invoked...

  // What we _want_ to test:
  //    - When our user clicks that button... what happens in the app?
  // it("displays a message when 'speak' button is clicked", () => {
  //   const speak = jest.fn();
  //   let messageMock = "";

  //   const { getByText } = render(<Speak speak={speak} message={messageMock} />);

  //   fireEvent.click(getByText(/speak/i));

  //   expect(speak).toHaveBeenCalled();
  // });
});

