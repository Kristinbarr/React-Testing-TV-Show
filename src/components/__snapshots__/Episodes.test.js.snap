// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Episodes /> snapshot matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="episodes"
  />
</DocumentFragment>
`;
